grafana_client/client.py

```diff
@@ -161,7 +161,8 @@ def __request_runner(url, json=None, data=None, headers=None):
             # The "Tempo" data source responds with text/plain.
             if r.headers.get("Content-Type", "").startswith("text/"):
                 return r.text
+            elif not r.text: # Fix for alert rules
+                return r.status_code
```

Comment:
Can you elaborate why this would be needed? The test cases you added are apparently also working without this change.

Reply:
On the other hand, when having it, it croaks on another test case:

```python
ERROR: test_grafana_client_no_verify (test.test_grafana_client.TestGrafanaClient)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/grafana-client/grafana-client/test/test_grafana_client.py", line 78, in test_grafana_client_no_verify
    grafana.users.find_user("test@example.org")
  File "/home/runner/work/grafana-client/grafana-client/grafana_client/elements/user.py", line 67, in find_user
    r = self.client.GET(search_user_path)
  File "/home/runner/work/grafana-client/grafana-client/grafana_client/client.py", line 164, in __request_runner
    elif not r.text:  # Fix for alert rules
AttributeError: 'MockResponse' object has no attribute 'text'
```

-- https://github.com/panodata/grafana-client/actions/runs/3269035697/jobs/5376128983#step:7:350

Reply:
Hello!
First of all I am glad you find this contribution useful.
answering your question…
it’s not easy to answer now, because I don’t remember after one month :( I believe it was failing because grafana API (9.0.1 or 9.0.0) didn’t respond with any body/text in case of adding contact point or alert rule, the only response was status code so that I had to do quick fix.

Reply:
All right, thanks. I would propose to merge your update without this specific change and re-evaluate if that is still happening with Grafana 9, and if that would really be needed. If so, then let us explicitly file it as a bug and submit a corresponding fix.